1-bootstrap/README.md

```diff
@@ -22,6 +22,166 @@ Each pipeline has the following associated resources:
+
+- Populate your `terraform.tfvars` file in `1-bootstrap` with the Cloud Build 2nd Gen configuration variable, here is an example:
+
+   ```hcl
+   cloudbuildv2_repository_config = {
+      repo_type = "GITHUBv2"
+
+      repositories = {
+         multitenant = {
+            repository_name = "eab-multitenant"
+            repository_url  = "https://github.com/your-org/eab-multitenant.git"
+         }
+
+         applicationfactory = {
+            repository_name = "eab-applicationfactory"
+            repository_url  = "https://github.com/your-org/eab-applicationfactory.git"
+         }
+
+         fleetscope = {
+            repository_name = "eab-fleetscope"
+            repository_url  = "https://github.com/your-org/eab-fleetscope.git"
+         }
+      }
+
+      github_secret_id                            = "projects/REPLACE_WITH_PRJ_NUMBER/secrets/github-pat" # Personal Access Token Secret
+      github_app_id_secret_id                     = "projects/REPLACE_WITH_PRJ_NUMBER/secrets/github-app-id" # App ID value secret
+   }
+   ```
+
+#### Cloud Build with Gitlab Pre-requisites
+
+To proceed with gitlab as your git provider you will need:
+
+- An authenticated Gitlab account. The steps in this documentation assumes you have a configured SSH key for cloning and modifying repositories.
+- A **private** GitLab repository for each one of the repositories below:
+  - Multitenant (`eab-multitenant`)
+  - Fleetscope (`eab-fleetscope`)
+  - Application Factory (`eab-applicationfactory`)
+
+   > Note: Default names for the repositories are, in sequence: `eab-multitenant`, `eab-fleetscope` and `eab-applicationfactory`; If you choose other names for your repository make sure you update `terraform.tfvars` the repository names under `cloudbuildv2_repository_config` variable.
+
+- An access token with the `api` scope to use for connecting and disconnecting repositories.
+
+- An access token with the `read_api` scope to ensure Cloud Build repositories can access source code in repositories.
+
+- Create a secret for the Gitlab API Access Token:
+
+   ```bash
+   GITLAB_API_TOKEN=<replace_with_app_id>
+   printf $GITLAB_API_TOKEN | gcloud secrets create gitlab-api-token --project=$GIT_SECRET_PROJECT --data-file=-
+   ```
+
+- Take note of the secret id, it will be used in `terraform.tfvars` later on:
+
+   ```bash
+   gcloud secrets describe gitlab-api-token --project=$GIT_SECRET_PROJECT --format="value(name)"
+   ```
+
+- Create a secret for the Gitlab Read API Access Token:
+
+   ```bash
+   GITLAB_READ_API_TOKEN=<replace_with_token>
+   printf $GITLAB_READ_API_TOKEN | gcloud secrets create gitlab-read-api-token --project=$GIT_SECRET_PROJECT --data-file=-
+   ```
+
+- Take note of the secret id, it will be used in `terraform.tfvars` later on:
+
+   ```bash
+   gcloud secrets describe gitlab-read-api-token --project=$GIT_SECRET_PROJECT --format="value(name)"
+   ```
+
+- Generate a random 36 character string that will be used as the Webhook Secret:
```

Comment:
I added the following note:

> NOTE: for testing purposes, you may use the following command to generate the webhook in bash: `GITLAB_WEBHOOK=$(cat /dev/urandom | tr -dc "[:alnum:]" | head -c 36)`

is that enough? I did this because the user might have their own way of securely creating a webhook secret

Reply:
Yup, looks graet to me.